src/bls12_381/ec.rs

```diff
@@ -1259,13 +1411,52 @@ pub mod g1 {
         assert_eq!(tmp2, c.into_projective());
     }
 
+    #[test]
+    fn test_g1_sw_encode_degenerate() {
+        // test the degenerate case t = 0
+        let p = G1Affine::sw_encode(Fq::zero());
+        assert!(p.is_on_curve());
+        assert!(p.is_zero());
+
+        // test the degenerate case t^2 = - b - 1
+        let mut t = Fq::one();
+        t.add_assign(&G1Affine::get_coeff_b());
+        t.negate();
+        let mut t = t.sqrt().unwrap();
+        let p = G1Affine::sw_encode(t);
+        assert!(p.is_on_curve());
+        assert!(!p.is_zero());
+        assert_eq!(p.y.parity(), t.parity());
```

Comment:
Oh, we should check `assert_eq!(p, G1::one())` since these are well-defined edge cases.